File: python/tstlp-sync.py
import sys
import os

def filenames_from_folder(path):
    filenames = []

    with os.scandir(path) as file_iterator:
        for dir_entry in file_iterator:
            filename = dir_entry.name
            filenames.append(filename)

    filenames.sort()

    return filenames

def count_string_in_file(path, s):
    counter = 0

    with open(path) as file:
        line = file.readline()

        while line:
            if s in line:
                counter += 1
            line = file.readline()

    return counter


def main():
    folder = sys.argv[1]

    for filename in filenames_from_folder(folder):
        stuckThreadCounter = count_string_in_file(folder + filename, "notifyStuckThreadDetected")
        print(filename + ":" + str(stuckThreadCounter))

main()

File: go/tstlp-sync.go
package main

import (
    "bufio"
    "fmt"
    "log"
    "os"
    "strings"
)

func main() {
    filenames := filenamesFromArgs()

    for _, filename := range filenames {
        countedStuckThread := countStuckThreadsInFile(filename);
        fmt.Printf("%s:%d\n", filename, countedStuckThread)
    }
}

func countStuckThreadsInFile(filename string) (int) {
    file, err := os.Open(filename)
    if err != nil {
        log.Fatal(err)
    }
    defer file.Close()

    counter := 0

    scanner := bufio.NewScanner(file)

    for scanner.Scan() {
        if strings.Contains(scanner.Text(), "notifyStuckThreadDetected") {
            counter++
        }
    }

    if err := scanner.Err(); err != nil {
        log.Fatal(err)
    }

    return counter
}

func filenamesFromArgs() ([]string) {
    args := os.Args[1:]

    if len(args) < 1 {
        fmt.Println("Error: No folder given")
        os.Exit(1)
    }

    folder := args[0]
    prefix := valueAtIndexOrDefault(args, 1, "")

    files, err := os.ReadDir(folder)
    if err != nil {
        log.Fatal(err)
    }

    var filenames []string

    for _, file := range files {
        if strings.Contains(file.Name(), prefix) {
            filenames = append(filenames, folder + "/" + file.Name())
        }
    }

    return filenames
}

func valueAtIndexOrDefault(args []string, index int, defaultValue string) (string) {
    if len(args) > index {
        return args[index]
    }
    return defaultValue
}
